### ISSUE TITLE
Support Client Sharding using Jedis Connector in Spring Session Redis

### ISSUE THREAD

**Issue by hk250097** [spring-projects/spring-session#2392]
**Expected Behavior**
We want to use the client side sharding using Jedis in Spring Session Redis.
Please refer https://github.com/redis/jedis/wiki/AdvancedUsage#shardedjedis for details.

Spring Session Redis uses Lettuce connector by default. Can we configure Jedis ?
The spring session is abstract , that is the client code do not deal with the data write or reading of data from redis.
As per https://github.com/redis/jedis/wiki/AdvancedUsage#shardedjedis we need to define the sharding at the client side.
Is there any feature supporting this or even the Lettuce sharding ?
How can we implement sharding in spring session ?
Is there anything in the feature release.
```
List<JedisShardInfo> shards = new ArrayList<JedisShardInfo>();
JedisShardInfo si = new JedisShardInfo("localhost", 6379);
si.setPassword("foobared");
shards.add(si);
si = new JedisShardInfo("localhost", 6380);
si.setPassword("foobared");
shards.add(si);
```

**Current Behavior**
We do not see client side sharding support in spring session ?
As client side requires to set the behaviour for shard, How can we implement or override the spring session to implement sharding.

**Context**

**Comment by marcusdacoregio** [comment]
Hi, @hk250097. I believe that this request should be made on the Spring Data Redis issue tracker.

See if https://github.com/spring-projects/spring-data-redis/issues/1007 and https://docs.spring.io/spring-data/data-redis/docs/current/reference/html/#cluster helps.

**Comment by marcusdacoregio** [comment]
Do you want to be able to configure the Jedis connection? If so, Spring Data Redis allows you to do that, it is the documentation that I linked above. If not, can you please share exactly what has to be done that is not possible now?

**Comment by hk250097** [comment]
@marcusdacoregio 
Here is the issue..in the current application we are using spring-session with Redis server.
The app gets lot of session (httpSession) data to be stored. When we migrated to micro service based architecture, we used spring-session with Redis to store the httpSession data with multiple servers (not sticky session)

There was issue in the production, so had to created 2 instances of Redis and do shrading at the cloudfare..i.e based on the logic it hits the redis1 and redis2 instance.

We are trying to remove that logic of cloud fare and have it in the application code to do the sharding.
Basically, lets say user1 logs in ...so the httpSession data should be put in redis1 and for user2 it should be put in redis2.
The application code should be handing it rather than the cloud fare.

The legacy (monolithic) code has lot of request.getSession . The spring session does internally uses redis to store this data.
I would want to override this functionality and when the code request.getSession gets executed, i want to put it in the redis1.
I would like to take control over and rather than spring-session doing it. The application should be able to decide where to put the session data. This should solve the cloudfare shrading and have this logic in application code.
